src/mcu.h

```diff
@@ -423,33 +508,16 @@ enum {
     ROM_SET_COUNT
 };
 
-extern const char* rs_name[ROM_SET_COUNT];
-
-extern int romset;
-
-extern int mcu_mk1;
-extern int mcu_cm300;
-extern int mcu_st;
-extern int mcu_jv880;
-extern int mcu_scb55;
-
-extern SDL_atomic_t mcu_button_pressed;
-
-static const uint32_t uart_buffer_size = 8192;
-extern uint32_t uart_write_ptr;
-extern uint32_t uart_read_ptr;
-extern uint8_t uart_buffer[uart_buffer_size];
-
-uint8_t MCU_ReadP0(void);
-uint8_t MCU_ReadP1(void);
-void MCU_WriteP0(uint8_t data);
-void MCU_WriteP1(uint8_t data);
-void MCU_GA_SetGAInt(int line, int value);
+uint8_t MCU_ReadP0(mcu_t& mcu);
+uint8_t MCU_ReadP1(mcu_t& mcu);
+void MCU_WriteP0(mcu_t& mcu, uint8_t data);
+void MCU_WriteP1(mcu_t& mcu, uint8_t data);
+void MCU_GA_SetGAInt(mcu_t& mcu, int line, int value);
 
-void MCU_EncoderTrigger(int dir);
+void MCU_EncoderTrigger(mcu_t& mcu, int dir);
 
-void MCU_PostSample(int *sample);
-void MCU_PostUART(uint8_t data);
+void MCU_PostSample(mcu_t& mcu, int *sample);
+void MCU_PostUART(mcu_t& mcu, uint8_t data);
 
-void MCU_WorkThread_Lock(void);
-void MCU_WorkThread_Unlock(void);
+void MCU_WorkThread_Lock(mcu_t& mcu);
+void MCU_WorkThread_Unlock(mcu_t& mcu);
```

Comment:
Not sure how you feel about const-correctness, but a lot of args and vars could be made `const`.